docs/search.md

```diff
@@ -0,0 +1,199 @@
+
+    class Django:
+        model = AnswerPage
+
+        fields = [
+            'search_tags',
+            'language',
+        ]
+```
+
+The fields in `fields` will be indexed without any preparation/manipulation, directly as are stored on the model. 
+
+### Custom fields
+
+Sometimes it might be desirable to index a field as an alternative type — say, the string that matches an integer for a Django field that species `choices`. 
+
+It might also be desirable to construct a field to index from multiple fields on the model, particular for Wagtail pages with stream fields. 
+
+We may also want to specify Elasticsearch-specific [field properties](https://elasticsearch-dsl.readthedocs.io/en/stable/persistence.html#mappings), like a custom analyzer.
+
+To do so, we specify [custom fields](https://django-elasticsearch-dsl.readthedocs.io/en/latest/fields.html) as attributes on the document class, with an `attr` argument that specifies the field on the model to reference.
+
+```python
+from django_elasticsearch_dsl import fields
+
+from search.elasticsearch_helpers import synonym_analyzer
+
+class AnswerPageDocument(Document):
+    text = fields.TextField(attr="text", analyzer=synonym_analyzer)
+```
+
+The `attr` on the model can be a `@property` or a Django model field.
+
+We can also do any data preparation/manipulation for fields using [`prepare_`-prefixed methods](https://django-elasticsearch-dsl.readthedocs.io/en/latest/fields.html#using-prepare-field).
+
+```python
+from django_elasticsearch_dsl import fields
+
+class AnswerPageDocument(Document):
+    portal_topics = fields.KeywordField()
+
+    def prepare_portal_topics(self, instance):
+        return [topic.heading for topic in instance.portal_topic.all()]
+```
+
+### Helpers
+
+We provide a few common helpers in `search.elasticsearch_helpers` for use in creating document classes:
+
+- `environment_specific_index(base_name)`: Generate the index name for the `base_name` that is specific to the deployment environment. This allows each index to be isolated to a deployment environment within an Elasticsearch cluster.
+- `label_autocomplete`: A reusable autocomplete analyzer for creating fields that [autocomplete terms](#autocomplete). This is used for type-ahead search boxes.
+- `synonym_analyzer`: A reusable analyzer for creating fields that will match synonyms of a search term.
+
+```python
+from search.elasticsearch_helpers import (
+    environment_specific_index,
+    label_autocomplete,
+    synonym_analyzer,
+)
+```
+
+### Building the index
+
+With the `Document` class created for your model in a `documents.py` module within a Django app listed in `INSTALLED_APPS`, all that is left to do is to use the [django-elasticsearch-dsl management commands](https://django-elasticsearch-dsl.readthedocs.io/en/latest/management.html) to rebuild the index:
+
+```shell
+./cfgov/manage.py search_index --create --models [app.Model]
+```
+
+The index can also be rebuilt at any time:
+
```

Comment:
```suggestion

```shell
./cfgov/manage.py search_index --rebuild -f --models [app.Model]
```

And all Django indexes can be rebuilt in parallel:

Reply:
Done in 4262e60!

Reply:
Thinking a bit more about our use of the `--models` arg, I think we'd routinely use the app name, not an individual model.

so I think the examples should use `[app]`, as in: `./cfgov/manage.py search_index --rebuild -f ask_cfpb`